### ISSUE TITLE
Add LAN8720 support for Raspberry Pico

### ISSUE THREAD

**Issue by ch3funio** [esphome/feature-requests#2078]
**Describe the problem you have/What new integration you would like**
LAN8720 module doesn't work with Raspberry Pico - fails during validation: `Component ethernet requires component esp32.`

**Please describe your use case for this integration and alternatives you've tried:**
Alternative: Raspberry Pico-W with wifi, but it's a no-go in this situation (no Wifi at the place).

**Additional context**

**Comment by BarryReid** [comment]
Did this ever get added?

**Comment by Meisam-kh** [comment]
I Agree with this feature. or add support with w5500 for Raspberry pi Pico.
